src/nvim/getchar.c

```diff
@@ -2250,23 +2250,26 @@ static int get_key_from_user(int *timedoutp, int *mode_deletedp,
       setcursor();                /* put cursor back where it belongs */
   }
 
-  if (bytes_read < 0)
-    return -1;                     /* end of input script reached */
-  if (bytes_read == NUL) {                 /* no character available */
+  if (bytes_read < 0) {
+    /* end of input script reached will return -1 below */
+  } else if (bytes_read == NUL) {
```

Comment:
I'd expect it to mean "number of bytes read", but if you're comparing it to `NUL`, that must not be true.